lib/msal-browser/src/utils/BrowserUtils.ts

```diff
@@ -11,146 +11,155 @@ import {
+        isIframedApp &&
+        !allowRedirectInIframe
+    ) {
+        // If we are not in top frame, we shouldn't redirect. This is also handled by the service.
+        throw createBrowserAuthError(BrowserAuthErrorCodes.redirectInIframe);
     }
+}
 
-    /**
-     * Block redirectUri loaded in popup from calling AcquireToken APIs
-     */
-    static blockAcquireTokenInPopups(): void {
-        // Popups opened by msal popup APIs are given a name that starts with "msal."
-        if (BrowserUtils.isInPopup()) {
-            throw createBrowserAuthError(
-                BrowserAuthErrorCodes.blockNestedPopups
-            );
-        }
+/**
+ * Block redirectUri loaded in popup from calling AcquireToken APIs
+ */
+export function blockAcquireTokenInPopups(): void {
+    // Popups opened by msal popup APIs are given a name that starts with "msal."
+    if (isInPopup()) {
+        throw createBrowserAuthError(BrowserAuthErrorCodes.blockNestedPopups);
     }
+}
 
-    /**
-     * Throws error if token requests are made in non-browser environment
-     * @param isBrowserEnvironment Flag indicating if environment is a browser.
-     */
-    static blockNonBrowserEnvironment(isBrowserEnvironment: boolean): void {
-        if (!isBrowserEnvironment) {
-            throw createBrowserAuthError(
-                BrowserAuthErrorCodes.nonBrowserEnvironment
-            );
-        }
+/**
+ * Throws error if token requests are made in non-browser environment
+ * @param isBrowserEnvironment Flag indicating if environment is a browser.
+ */
+export function blockNonBrowserEnvironment(
+    isBrowserEnvironment: boolean
+): void {
+    if (!isBrowserEnvironment) {
+        throw createBrowserAuthError(
+            BrowserAuthErrorCodes.nonBrowserEnvironment
+        );
     }
+}
 
-    /**
-     * Throws error if initialize hasn't been called
-     * @param initialized
-     */
-    static blockAPICallsBeforeInitialize(initialized: boolean): void {
-        if (!initialized) {
-            throw createBrowserAuthError(
-                BrowserAuthErrorCodes.uninitializedPublicClientApplication
-            );
-        }
+/**
+ * Throws error if initialize hasn't been called
+ * @param initialized
+ */
+export function blockAPICallsBeforeInitialize(initialized: boolean): void {
+    if (!initialized) {
+        throw createBrowserAuthError(
+            BrowserAuthErrorCodes.uninitializedPublicClientApplication
+        );
     }
 }
+
+/**
+ * Adds a preconnect link element to the header which begins DNS resolution and SSL connection in anticipation of the /token request
+ * @param loginDomain Authority domain, including https protocol e.g. https://login.microsoftonline.com
+ * @returns
+ */
+export function preconnect(authority: string): void {
+    const link = document.createElement("link");
+    link.rel = "preconnect";
+    link.href = new URL(authority).origin;
+    link.crossOrigin = "anonymous";
```

Comment:
because l.mso origin is different from the SPA url?

Reply:
That's right